src/hub.js

```diff
@@ -245,17 +252,45 @@ function mountUI(scene) {
     document.getElementById("ui-root")
   );
 
-  getAvailableVREntryTypes().then(availableVREntryTypes => {
-    uiRoot.setState({ availableVREntryTypes });
-    uiRoot.handleForcedVREntryType();
-  });
+  return uiRoot;
 }
 
-const onReady = () => {
+const onReady = async () => {
   const scene = document.querySelector("a-scene");
   document.querySelector("a-scene canvas").classList.add("blurred");
   window.APP.scene = scene;
-  mountUI(scene);
+
+  const uiRoot = mountUI(scene);
+
+  getAvailableVREntryTypes().then(availableVREntryTypes => {
+    uiRoot.setState({ availableVREntryTypes });
+    uiRoot.handleForcedVREntryType();
+  });
+
+  const environmentRoot = document.querySelector("#environment-root");
+
+  const initialEnvironmentEl = document.createElement("a-entity");
+  initialEnvironmentEl.addEventListener("bundleloaded", () => uiRoot.setState({ initialEnvironmentLoaded: true }));
+  environmentRoot.appendChild(initialEnvironmentEl);
+
+  if (qs.room) {
+    // If ?room is set, this is `yarn start`, so just use a default environment and query string room.
+    uiRoot.setState({ janusRoomId: qs.room && !isNaN(parseInt(qs.room)) ? parseInt(qs.room) : 1 });
+    initialEnvironmentEl.setAttribute("gltf-bundle", "src: /assets/environments/cliff_meeting_space/bundle.json");
+    return;
+  }
+
+  const hubId = document.location.pathname.substring(1).split("/")[0];
+  console.log(`Hub ID: ${hubId}`);
+  const res = await fetch(`/api/v1/hubs/${hubId}`);
+  const data = await res.json();
+  const hub = data.hubs[0];
+  const defaultSpaceChannel = hub.channels.find(c =>
+    c.attributes.find(a => a.length === 1 && a[0] === "default-space")
```

Comment:
yeah I can probably dial back the format here, its probably overengineering. basically i'm going on "past experience" with having arbitrary named key/value tag pairs for the space. so this is [["default-space"]] which is effectively a attribute with no value (otherwise the array would have two elements. we could change that to a map, or drop it altogether.

i think hubs[0] is fine because the API should eventually support fetching multiple hubs, so there's a wrapper `hubs` key that has an array attached. not sure if there is a more idiomatic way to do that these days but this is typically how I deal with future proofing API response going from 1 -> N

Reply:
Fair enough, `hubs[0]` makes sense.